app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -988,43 +988,29 @@ public function removeFromLists($lead, $lists, $manuallyRemoved = true)
 
     /**
      * Add lead to Stage.
-     *
-     * @param array|Lead  $lead
-     * @param array|Stage $stage
-     * @param bool        $manuallyAdded
-     *
-     * @return $this
      */
-    public function addToStages($lead, $stage, $manuallyAdded = true)
+    public function addToStage(Lead $lead, Stage $stage, string $origin): LeadModel
```

Comment:
Changing a public/protected method signature is a BC break. Please create a new method instead and mark this one as deprecated. Same applies to other methods that change signature.